### PR TITLE
fix: NumberField increment/decrement buttons no longer update display value

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 📖 Description

A [recent change](https://github.com/microsoft/fast/pull/5205) to the NumberField inadvertently broke the increment/decrement buttons from changing the displayed value when pressed. More broadly, any programmatic change of the value will not update the displayed value appropriately.

### 🎫 Issues

#5246 

## 📑 Test Plan

I've added a couple of unit tests that focus on the `stepUp` and `stepDown` methods updating the displayText as expected. While the issue is technically broader than this, this seems like a reasonable set of tests.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by atmgrifter00** [comment]
I'm not sure why I had to delete that other change file, but it was present after my fork, and `yarn change` complained about its presence.

File: packages/web-components/fast-foundation/src/number-field/number-field.ts [hunk 1]
```diff
@@ -209,6 +209,7 @@ export class NumberField extends FormAssociatedNumberField {
         }
 
         if (value != this.value) {
+            this.displayText = value;
```
Comment (robarbms) [inline packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: This won't work. It breaks typing in values that aren't valid while you type them. Such as trying to type **-.5e10** isn't possible with this. We'd have to make adjustments to displayText in stepUp and stepDown methods.

We could update the displayText in **valueChanged()** to catch changes to value by script, but that could update it during the **updateValue()** method when we don't want it to.

One way to deal with it all is to create a flag, **updatingValue** that is set to true when **updateValue()** is called. It would set it to false after it is finished. Then we could just set **this.displayText** to **this.value** when **this.updatingValue** is equal to false. This would update when changed by script and would fix **stepUp** and **stepDown**.

**Review by robarbms — CHANGES_REQUESTED** [review]

File: packages/web-components/fast-foundation/src/number-field/number-field.ts (hunk 1 above)
Comment (atmgrifter00) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: Cool...you want me to try and tackle this?

Comment (robarbms) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: I can get this. It would fix updating with scripts and increment and decrement. I'd like to build more tests around this to catch all of these edge cases as well.

Comment (atmgrifter00) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: Also, just so there's no confusion, my change does not seem to inhibit the text entry you outlined.

Comment (robarbms) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: It was lagging and would sometimes replace the value with 0 when I was playing around with entering numbers.

Comment (atmgrifter00) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: Ah, I see. Yeah, after playing with it a bit, the input does seem a little more fussy with my change present (not impossible, but definitely fussier). Still, I feel that the inevitable fix for [this Issue](https://github.com/microsoft/fast/issues/5249) might address that.

Comment (robarbms) [reply packages/web-components/fast-foundation/src/number-field/number-field.ts:212]: I created a number-field with a min="10". With the new displayText property, it won't coerce the value to the min while typing. The internal value will actually be 10, but it will show what the user is typing now.
